Tools/scripts/build_tests/test_ccache.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# test ccache efficiency building two similar boards
+
+import subprocess
+import re
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(description='test ccache performance')
+parser.add_argument('file', nargs='?', default=None, help='bin log file containing raw IMU logs')
+parser.add_argument('--boards', default='MatekF405,MatekF405-bdshot', help='boards to test')
+parser.add_argument('--min-cache-pct', type=int, default=75, help='minimum ccache hit rate')
+
+args = parser.parse_args()
+
+def ccache_stats():
+    '''return hits/misses from ccache -s'''
+    hits = 0
+    miss = 0
+    stats=str(subprocess.Popen(["ccache", "-s"],stdout=subprocess.PIPE).communicate()[0],encoding='ascii')
+    for line in stats.split('\n'):
+        m=re.match("cache.hit\D*(\d+)$", line)
+        if m is not None:
+            hits += int(m.group(1))
+        m=re.match("cache.miss\D*(\d+)", line)
+        if m is not None:
+            miss += int(m.group(1))
+    return (hits,miss)
+
+def build_board(boardname):
+    subprocess.run(["./waf", "configure", "--board", boardname])
+    subprocess.run(["./waf", "clean", "copter"])
+
+boards = args.boards.split(",")
+if len(boards) != 2:
+    print(boards)
+    print("Must specify exactly 2 boards (comma separated)")
+    sys.exit(1)
+
+subprocess.run(["ccache", "-C", "-z"])
```

Comment:
Maybe set `CACHE_BASEDIR` in the environment rather than killing the user's ccache?

Reply:
done (CCACHE_DIR)